src/main/java/net/sf/jabref/gui/worker/LookupDOIsWorker.java

```diff
@@ -0,0 +1,69 @@
+package net.sf.jabref.gui.worker;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import net.sf.jabref.gui.BasePanel;
+import net.sf.jabref.gui.JabRefFrame;
+import net.sf.jabref.gui.undo.NamedCompound;
+import net.sf.jabref.gui.undo.UndoableFieldChange;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.logic.util.DOI;
+import net.sf.jabref.model.FieldChange;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class LookupDOIsWorker extends AbstractWorker {
+
+    private final JabRefFrame frame;
+    private String message;
+
+    private static final Log LOGGER = LogFactory.getLog(LookupDOIsWorker.class);
+
+    public LookupDOIsWorker(JabRefFrame frame) {
```

Comment:
For maximal reusability, could you please make this class independent on DOI. There are many other identifier for which such an automatic look-up makes sense.